feat(stackit wiab): WPB-14388 improving stackIT-wiab documentation fo… (#757)
wireapp/wire-server-deploy: offline/stackIT-wiab.md

| Any      | egress    | Any        | Any      | IPv4       | Any        | Allow all outgoing IPv4 traffic             |
| Any      | egress    | Any        | Any      | IPv6       | Any        | Allow all outgoing IPv6 traffic             |
| tcp      | ingress   | 22         | 22       | IPv4       | 0.0.0.0/0  | Allow SSH access                            |
| tcp      | ingress   | 443        | 443      | IPv4       | 0.0.0.0/0  | Allow HTTPS traffic                         |
| tcp      | ingress   | 80         | 80       | IPv4       | 0.0.0.0/0  | Allow HTTP traffic                          |
| tcp      | ingress   | 3478       | 3478     | IPv4       | 0.0.0.0/0  | Allow alternative STUN/TURN traffic over TCP|
| udp      | ingress   | 3478       | 3478     | IPv4       | Any        | Allow STUN/TURN traffic for Coturn          |
| udp      | ingress   | 32768      | 61000    | IPv4       | 0.0.0.0/0  | Allow calling traffic for Coturn over UDP   |

## Steps to Deploy WIAB from local environment (or on stackIT node)

### 1. Clone the repository
   - `git clone https://github.com/wireapp/wire-server-deploy.git`
   - `cd wire-server-deploy`

> ### 2. Prepare the variables for wire deployment

Reviewer: ```suggestion
### 2. Prepare the variables for Wire deployment
```

- Prepare DNS records, StackIT public IP and set up Cert Manager (for example, letsencrypt) to start before next step as mentioned [here](https://docs.wire.com/how-to/install/helm.html#how-to-set-up-dns-records).
   - Check file `stackIT/host.ini` for host details, replace example.com with the host machine.

>    - Check file `stackIT/stackit-vm-setup.yml` to define target_domain, replace example.com with the wire host domain - Ansible tasks will take care of other replacement operations.

Reviewer: ```suggestion
   - Check file `stackIT/stackit-vm-setup.yml` to define target_domain, replace example.com with the desired base domain of your Wire deployment - Ansible tasks will take care of other replacement operations.
```

  ```

- **The above command will accomplish the following tasks:**
  - Deploy a Minikube Kubernetes cluster using docker containers as base, and 4 Docker containers to support assethost and datastore requirements. The functionality of different nodes is explained [here](https://docs.wire.com/how-to/install/planning.html#production-installation-persistent-data-high-availability).
  - Generate `hosts.ini` based on the IPs of above containers for further ansible operations on containers. Read more [here](https://github.com/wireapp/wire-server-deploy/blob/master/offline/docs_ubuntu_22.04.md#example-hostsini).
  - Download wire-server-deploy artifacts in the user's home directory. Read more [here](https://github.com/wireapp/wire-server-deploy/blob/master/offline/docs_ubuntu_22.04.md#artifacts-provided-in-the-deployment-tarball)
  - Configure iptables rules to handle the traffic for k8s Nginx Controller and handle DNAT for Coturn  (used by Wire applications). Read more [here](https://github.com/wireapp/wire-server-deploy/blob/master/offline/docs_ubuntu_22.04.md#directing-traffic-to-wire).

---

### 3. Commands to Run on the StackIT Node in directory `wire-server-deploy`

> #### Note: These commands can be collected to run inside a single script, here we have broken down the deployment into small collective steps. These commands can work only from the stackIT node and directory wire-server-deploy.

Reviewer: ```suggestion
#### Note: These commands can be collected to run inside a single script, here we have broken down the deployment into small collective steps. These commands can work only from the stackIT node and in the directory wire-server-deploy.
```
